### PR TITLE
Implement subgroups endpoint

### LINKED ISSUE #295: Subgroups api missing or not readily visible
I've been trying to get the subgroups for a group, however I don't see any other way than to get all groups and then filter by parent id. This seems a bit roundabout, seeing how gitlab does offer a list subgroup endpoint:

https://docs.gitlab.com/ee/api/groups.html#list-a-groups-subgroups

This endpoint is basically a scoped groups endpoint.
This plus the list descendant groups would be nice additions. We heavily structure our projects in subgroups within subgroups so having something to work with that would be nice.

### PR DESCRIPTION
**Pull request by Rimgelair**
Fix https://github.com/ubisoft/NGitLab/issues/295

### CODE REVIEW

File: NGitLab/Impl/GroupsClient.cs [hunk 1]
```diff
@@ -112,6 +117,18 @@ public Task<Group> GetByFullPathAsync(string fullPath, CancellationToken cancell
             return _api.Get().ToAsync<Group>(Url + "/" + Uri.EscapeDataString(fullPath), cancellationToken);
         }
 
+        public GitLabCollectionResponse<Group> GetSubgroupsByIdAsync(int id, GroupQuery query)
```
Comment (meziantou) [inline NGitLab/Impl/GroupsClient.cs]: I think query should allow null and be optional. Something like 

```suggestion
        public GitLabCollectionResponse<Group> GetSubgroupsByIdAsync(int id, GroupQuery query = null)
```

File: NGitLab/IGroupsClient.cs [hunk 2]
```diff
@@ -36,6 +36,10 @@ public interface IGroupsClient
 
         Task<Group> GetByFullPathAsync(string fullPath, CancellationToken cancellationToken = default);
 
+        GitLabCollectionResponse<Group> GetSubgroupsByIdAsync(int id, GroupQuery query);
```
Comment (meziantou) [inline NGitLab/IGroupsClient.cs]: You can't reuse `GroupQuery` here. The available options are not the same (e.g. `top_level_only`). You must create new class `SubgroupQuery`.

**Review by meziantou — CHANGES_REQUESTED** [review]

File: NGitLab/Impl/GroupsClient.cs (hunk 1 above)
Comment (Rimgelair) [reply NGitLab/Impl/GroupsClient.cs]: I agree I have made the changes.

File: NGitLab/IGroupsClient.cs (hunk 2 above)
Comment (Rimgelair) [reply NGitLab/IGroupsClient.cs]: That was an oversight haha, I only checked the GroupQuery class when I should have cross referenced with he docs. I have made the change.

File: NGitLab/Models/SubgroupQuery.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    /// Allows to use more advanced GitLab queries for getting subgroups (based on v4 GitLab API).
+    /// https://docs.gitlab.com/ee/api/groups.html#list-a-groups-subgroups
+    /// </summary>
+    public class SubgroupQuery
+    {
+        /// <summary>
+        /// Skip the group IDs passed
+        /// (optional)
+        /// </summary>
+        public int[] SkipGroups;
```
Comment (meziantou) [inline NGitLab/Models/SubgroupQuery.cs]: Use properties everywhere (using fields is _legacy_)

```suggestion
        public int[] SkipGroups { get; set; }
```

**Review by meziantou — CHANGES_REQUESTED** [review]

File: NGitLab/Models/SubgroupQuery.cs (hunk 3 above)
Comment (Rimgelair) [reply NGitLab/Models/SubgroupQuery.cs]: Done 😄

**Review by meziantou — APPROVED** [review]

**Pull request merged**